### DISCUSSION TITLE
Feature request: TileDB support

### DISCUSSION THREAD

**Discussion by RaananHadar** [developmentseed/titiler#415] [Feature Request]
[TileDB](https://tiledb.com/) is a next gen data storage engine with excellent s3 and gdal support. While not as ubiquitous as COG, it has many advantages.

As TiTiler is a very easy to use, being able to use TiTiler to open TileDB raster data would be very useful. I hope you consider this.

**Comment by vincentsarago** [comment]
@RaananHadar, TileDB has direct support in GDAL but it's an optional driver (need custom compilation). If you compile GDAL with TileDB support and then install rasterio pointing to your GDAL source (`pip install titiler.application --no-binary rasterio`) you would have TileDB support in titiler.

TiTiler itself is file format agnostic and rely on drivers available in your rasterio/gdal installation

**Reply by rowanwins** [reply]
TileDB stored arrays typically don't have overviews (although there are probably some workarounds) so I'm not sure how well it would perform compared to storing your data in COGs...

**Comment by RaananHadar** [comment]
That's a legitimate point. I know that tiledb has map server integration. So I'll check on their slack for best practices. 

I still think that this can be useful for interactive exploratory data analysis.
